src/main/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategy.java

```diff
@@ -76,15 +82,38 @@ private void done(Executor executor) {
         }
         LOGGER.log(Level.FINE, "terminating {0} since {1} seems to be finished", new Object[] {c.getName(), exec});
         c.setAcceptingTasks(false); // just in case
-        // Best to kill them off ASAP; otherwise NodeProvisioner does nothing until ComputerRetentionWork has run, causing poor throughput:
-        try {
-            c.getNode().terminate();
-        } catch (InterruptedException x) {
-            LOGGER.log(Level.WARNING, null, x);
-        } catch (IOException x) {
-            LOGGER.log(Level.WARNING, null, x);
+        synchronized (this) {
+            if (terminating) {
+                return;
+            }
+            terminating = true;
```

Comment:
correct we want the lock here to ensure exclusivity of starting the background task